webui-streamlit.cmd

```diff
@@ -3,59 +3,72 @@
 :: Run all commands using this script's directory as the working directory
 cd %~dp0
 
-:: copy over the first line from environment.yaml, e.g. name: ldm, and take the second word after splitting by ":" delimiter 
-set /p first_line=< environment.yaml  
-for /f "tokens=2 delims=:" %%i in ("%first_line%") do set untrimmed_conda_env_name=%%i
-for /f "tokens=* delims= " %%a in ("%untrimmed_conda_env_name%") do set conda_env_name=%%a
-echo Environment name is set as %conda_env_name% as per environment.yaml
+:: copy over the first line from environment.yaml, e.g. name: ldm, and take the second word after splitting by ":" delimiter
+for /F "tokens=2 delims=: " %%i in (environment.yaml) DO (
+  set v_conda_env_name=%%i
+  goto EOL
+)
+:EOL
 
-:: Put the path to conda directory after "=" sign if it's installed at non-standard path:
-set custom_conda_path=
+echo Environment name is set as %v_conda_env_name% as per environment.yaml
 
-IF NOT "%custom_conda_path%"=="" (
-  set paths=%custom_conda_path%;%paths%
+:: Put the path to conda directory in a file called "custom-conda-path.txt" if it's installed at non-standard path
+IF EXIST custom-conda-path.txt (
+  FOR /F %%i IN (custom-conda-path.txt) DO set v_custom_path=%%i
 )
-:: Put the path to conda directory in a file called "custom-conda-path.txt" if it's installed at non-standard path:
-FOR /F %%i IN (custom-conda-path.txt) DO set custom_conda_path=%%i
 
-set paths=%ProgramData%\miniconda3
-set paths=%paths%;%USERPROFILE%\miniconda3
-set paths=%paths%;%ProgramData%\anaconda3
-set paths=%paths%;%USERPROFILE%\anaconda3
+set v_paths=%ProgramData%\miniconda3
+set v_paths=%v_paths%;%USERPROFILE%\miniconda3
+set v_paths=%v_paths%;%ProgramData%\anaconda3
+set v_paths=%v_paths%;%USERPROFILE%\anaconda3
 
-for %%a in (%paths%) do (
- IF NOT "%custom_conda_path%"=="" (
-   set paths=%custom_conda_path%;%paths%
+for %%a in (%v_paths%) do (
+ IF NOT "%v_custom_path%"=="" (
+   set v_paths=%v_custom_path%;%v_paths%
  )
 )
 
-for %%a in (%paths%) do (
+for %%a in (%v_paths%) do (
  if EXIST "%%a\Scripts\activate.bat" (
-    SET CONDA_PATH=%%a
+    SET v_conda_path=%%a
     echo anaconda3/miniconda3 detected in %%a
-    goto :foundPath
+    goto :CONDA_FOUND
  )
 )
 
-IF "%CONDA_PATH%"=="" (
+IF "%v_conda_path%"=="" (
   echo anaconda3/miniconda3 not found. Install from here https://docs.conda.io/en/latest/miniconda.html
   exit /b 1
 )
 
+:CONDA_FOUND
+echo Stashing local changes and pulling latest update...
 call git stash
 call git pull
+call "%v_conda_path%\Scripts\activate.bat"
+
+for /f "delims=" %%a in ('git log -1 --format^="%%H" -- environment.yaml')  DO set v_cur_hash=%%a
+set /p "v_last_hash="<"z_version_env.tmp" 1>NUL 2>&1
```

Comment:
I accidentally left some test error output redirection here which isn't working. So going to remove it